resources/aws_ec2.go

```diff
@@ -680,10 +691,25 @@ func (obj *AwsEc2Res) CheckApply(apply bool) (checkOK bool, err error) {
 	}
 
 	if len(diOutput.Reservations) < 1 && obj.State == "terminated" {
+		// clear any IP addresses we've stored for send/recv.
```

Comment:
There is a bug in the code (or this patch) because if `len(diOutput.Reservations) > 1 && !apply, then you'll return from CheckApply without ever having touched this value!

So where's the bug? Either it's in this patch since you forgot about that case, OR having a `len(diOutput.Reservations) > 1` is unexpected and unhandled... Meaning either it should be an error or dealt with somehow, etc...